### PR TITLE
React 16.3 and Demo

### PR DESCRIPTION
**Pull request by chrisdakin**
Fixes [CS-16356](https://jira.archoninfosys.com:8181/browse/CS-16356?filter=-1)
Related: [Civicsource #1657](https://github.com/civicsource/civicsource/pull/1657)

Issue: In preparation for react 17, various lifecycle methods needed to be removed.

Fix: Move contents of componentWillReceiveProps and componentWillMount to componentDidUpdate.

Also adds an example that can be run locally.

### CODE REVIEW

**Comment by chrisdakin** [comment]
@chadly I've never setup webpack config stuff so there might be (read: is) a better way to run the example locally (like with hot reload or whatever magic stuff), not sure exactly what you want that setup to look like though.

Also not sure if we need the packagelock commited, or if we want to go with npm vs yarn, etc.

**Review by mdarens — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -4,7 +4,9 @@
   "description": "A simple react component that allows you to get a confirmation of action from the user before executing an action.",
   "main": "index.js",
   "scripts": {
-    "prepublish": "babel index.jsx -o index.js"
+    "prepublish": "babel index.jsx -o index.js",
+    "build": "webpack --progress --bail",
```
Comment (chadly) [inline package.json]: We don't need this `build` script.

**Review by chadly — CHANGES_REQUESTED** [review]
Use yarn & webpack v4

File: package.json [hunk 2]
```diff
@@ -29,16 +31,21 @@
     ]
   },
   "peerDependencies": {
-    "react": "15.x || 16.x"
+    "react": "^16.3"
```
Comment (chadly) [inline package.json]: Does the ^ allow lower than 16.3? I think it should be `>=16.3 < 18`.

**Review by chadly — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -29,16 +31,22 @@
...
+    "babel-preset-stage-2": "^7.0.0-beta.3",
+    "webpack": "4.x",
+    "webpack-cli": "^1.5.1",
+    "webpack-dev-server": "3.x"
   },
   "dependencies": {
-    "prop-types": "^15.5.10"
+    "prop-types": "^15.5.10",
+    "react": "^>=16.3 < 18",
+    "react-dom": "^>=16.3 < 18"
```
Comment (chadly) [inline package.json]: Don't add `react`/`react-dom` as dependencies. We want to keep them as `peerDependencies` to let consuming apps not accidentally pull in multiple react versions.

To make the demo work, move these to `devDependencies`.

**Review by chadly — CHANGES_REQUESTED** [review]

**Review by mdarens — DISMISSED** [review]

File: package.json [hunk 4]
```diff
@@ -40,13 +39,13 @@
     "babel-preset-es2015": "^7.0.0-beta.3",
     "babel-preset-react": "^7.0.0-beta.3",
     "babel-preset-stage-2": "^7.0.0-beta.3",
+    "react": "^>=16.3 < 18",
+    "react-dom": "^>=16.3 < 18",
```
Comment (chadly) [inline package.json]: For `devDependencies`, you can just use the specific version: `^16.3.0`. The range is for the consuming app.

**Review by chadly — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -29,14 +30,20 @@
     ]
   },
   "peerDependencies": {
-    "react": "15.x || 16.x"
+    "react": "^>=16.3 < 18"
```
Comment (chadly) [inline package.json]: The `^` is not valid methinks

**Review by chadly — CHANGES_REQUESTED** [review]

**Review by chadly — APPROVED** [review]

**Comment by chadly** [comment]
Published as [`v3.0.0`](https://github.com/civicsource/react-inline-confirm/releases/tag/v3.0.0)

**Pull request merged**